docs/backup-restore-integration.md

```diff
@@ -0,0 +1,334 @@
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vm1
+  namespace: ns1
+...
+```
+
+- ("kubevirt.io", "VirtualMachine", "ns1", "vm1")
+
+#### spec.instancetype
+
+```yaml
+...
+spec:
+  instancetype:
+    kind: VirtualMachineInstancetype
+    name: small
+...
+```
+
+- ("kubevirt.io", "VirtualMachineInstancetype", "ns1", "small")
+
+```yaml
+...
+spec:
+  instancetype:
+    kind: VirtualMachineClusterInstancetype
+    name: csmall
+...
+```
+
+- ("kubevirt.io", "VirtualMachineClusterInstancetype", "", "csmall")
+
+#### spec.preference
+
+```yaml
+...
+spec:
+  preference:
+    kind: VirtualMachinePreference
+    name: windows
+...
+```
+
+- ("kubevirt.io", "VirtualMachineClusterPreference", "ns1", "windows")
+
+```yaml
+...
+spec:
+  preference:
+    kind: VirtualMachineClusterPreference
+    name: cwindows
+...
+```
+
+- ("kubevirt.io", "VirtualMachineClusterPreference", "", "cwindows")
+
+#### spec.template
+
+See [VirtualMachineInstance](#virtualmachineinstance-object-graph)
+
+### VirtualMachineInstance Object Graph
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: vmi1
+  namespace: ns1
+...
+```
+
+- ("kubevirt.io", "VirtualMachineInstance", "ns1", "vmi1")
+- ("", "Pod", "ns1", "virt-launcher-vmi1-XXXXX") \*
+
+\* Each VirtualMachineInstance has a corresponding uniquely named Pod.  The backup process can look up the name of this pod by using the `kubevirt.io/created-by=<UID of VirtualMachineInstance>` label selector.
+
+#### spec.priorityClassName
+
+```yaml
+...
+spec:
+  priorityClassName: pc1
+...
+```
+
+- ("scheduling.k8s.io", "PriorityClass", "", "pc1")
+
+#### spec.volumes[\*].persistentVolumeClaim
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    persistentVolumeClaim:
+      claimName: pvc1
+...
+```
+
+- ("", "PersistentVolumeClaim", "ns1", "pvc1")
+- ("cdi.kubevirt.io", "DataVolume", "ns1", "pvc1")
+
+#### spec.volumes[\*].dataVolume
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    dataVolume:
+      name: dv1
+...
+```
+
+- ("", "PersistentVolumeClaim", "ns1", "dv1")
+- ("cdi.kubevirt.io", "DataVolume", "ns1", "dv1")
+
+#### spec.volumes[\*].configMap
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    configMap:
+      name: cm1
+...
+```
+
+- ("", "ConfigMap", "ns1", "cm1")
+
+#### spec.volumes[\*].secret
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    secret:
+      name: s1
+...
+```
+
+- ("", "Secret", "ns1", "s1")
+
+#### spec.volumes[\*].serviceAccount
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    serviceAccount:
+      serviceAccountName: sa1
+...
+```
+
+- ("", "ServiceAccount", "ns1", "sa1")
+
+#### spec.volumes[\*].memoryDump
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    memoryDump:
+      claimName: pvc1
+...
+```
+
+- ("", "PersistentVolumeClaim", "ns1", "pvc1")
+
+#### spec.networks[\*].multus
+
+```yaml
+...
+spec:
+  networks:
+  - name: ovs-net
+    multus:
+      networkName: ovs-vlan-100
```

Comment:
from what I understand the namespace of the multus network doesnt have to be the same as the vmi according to the comment: 
// References to a NetworkAttachmentDefinition CRD object. Format:
// <networkName>, <namespace>/<networkName>. If namespace is not
// specified, VMI namespace is assumed.
not sure if should be mentioned, or the doc doesnt come down to this level of details

Reply:
After talking we a Network person he said that he believes that network should not be backported and restored together with the vm. the network is something that is not necessarily 1:1 with a vm and usually will be used by other vms either at the same namespace or a dedicated network namespace. restoring a network can cause a bunch of problems. It seems like if one likes to restore a vm it needs to make sure it has the appropriate network exists when doing so.

Reply:
okay they will not be added to the graph.